WiserTaskScheduler/WiserTaskScheduler/Core/Services/CleanupService.cs

```diff
@@ -112,6 +113,50 @@ private async Task CleanupDatabaseLogsAsync(IDatabaseConnection databaseConnecti
         }
     }
 
+    /// <summary>
+    /// Cleanup files in the wiser_itemfile table with property_name 'TEMPORARY_FILE_FROM_WISER' older than 24h.
+    /// Note: this does not yet support dedicated tables.
+    /// </summary>
+    /// <param name="databaseConnection">The database connection to use.</param>
+    /// <param name="databaseHelpersService">The <see cref="IDatabaseHelpersService"/> to use.</param>
+    private async Task CleanupTemporaryWiserFilesAsync(IDatabaseConnection databaseConnection, IDatabaseHelpersService databaseHelpersService)
+    {
+        try
+        {
+            // Check if the table exists before trying to delete from it.
+            if (!await databaseHelpersService.TableExistsAsync(WiserTableNames.WiserItemFile))
+            {
+                return;
+            }
+
+            // Check if the property name is set and not just whitespace(or is empty).
+            if (!string.IsNullOrWhiteSpace(cleanupServiceSettings.ProperyNameTemporaryWiserFiles))
+            {
+                return;
+            }
+
+            // Check if the number of days to store is greater than 0.
+            if (cleanupServiceSettings.NumberOfDaysToStoreTemporaryWiserFiles <= 0)
+            {
+                return;
+            }
+
+            databaseConnection.AddParameter("propertyName", cleanupServiceSettings.ProperyNameTemporaryWiserFiles);
+            databaseConnection.AddParameter("cleanupDate", DateTime.Now.AddDays(-cleanupServiceSettings.NumberOfDaysToStoreTemporaryWiserFiles));
+            var rowsDeleted = await databaseConnection.ExecuteAsync($"DELETE FROM {WiserTableNames.WiserItemFile} WHERE property_name = ?propertyName AND added_on < ?cleanupDate", cleanUp: true);
```

Comment:
It might be better to get the IDs first, then delete the rows based on those IDs. We've found that that is much faster, for some reason, in MySQL.